src/util/worker/mod.rs

```diff
@@ -275,22 +283,87 @@ where
     runner.shutdown();
 }
 
-// Fill buffer with next task batch comes from `rx`.
-fn fill_task_batch<T>(rx: &Receiver<Option<T>>, buffer: &mut Vec<T>, batch_size: usize) -> bool {
-    match rx.recv() {
-        Ok(Some(task)) => {
-            buffer.push(task);
-            while buffer.len() < batch_size {
-                match rx.try_recv() {
-                    Ok(Some(t)) => buffer.push(t),
-                    Err(TryRecvError::Empty) => break,
-                    Ok(None) | Err(_) => return false,
+fn poll_with_timer<R, T, U>(
+    mut runner: R,
+    rx: Receiver<Option<T>>,
+    counter: Arc<AtomicUsize>,
+    batch_size: usize,
+    mut timer: Timer<U>,
+) where
+    R: BatchRunnableWithTimer<T, U> + Send + 'static,
+    T: Display + Send + 'static,
+    U: Send + 'static,
+{
+    let name = thread::current().name().unwrap().to_owned();
+    let mut batch = Vec::with_capacity(batch_size);
+    let mut keep_going = true;
+    let mut tick_time = None;
+    while keep_going {
+        tick_time = tick_time.or_else(|| timer.next_timeout());
```

Comment:
I am confused to use this, why not using timer next_timeout later directly?

Reply:
Because we don't need to trigger `timeout task` after we get a batch of tasks. @BusyJay

Reply:
still confused for this optimization. Are you afraid that calling next_timeout can reduce the performance?

Reply:
It's for this case: we get a timeout `3` seconds, then we fetch some tasks after `2` seconds. So next time when we want a `timeout` in `recv_timout`, it will be `1` second instead of a new value from `timer.next_timeout()`.

Reply:
but we will first call recv_timeout then fetch the tasks, and fetching the task may take a very long time?

Reply:
I suggest making thing simple and easy to understand.